test/plugin/test_in_forward.rb

```diff
@@ -583,5 +583,53 @@ def send_data(data, try_to_receive_response=false, response_timeout=1)
     @responses << res if try_to_receive_response
   end
 
+  # TODO: Use sub_test_case. Currently Errno::EADDRINUSE happens inside sub_test_case
+  test 'message protocol with source_hostname_key' do
+    execute_test { |events|
+      events.each { |tag, time, record|
+        send_data [tag, time, record].to_msgpack
+      }
+    }
+  end
+
+  test 'forward protocol with source_hostname_key' do
+    execute_test { |events|
+      entries = []
+      events.each {|tag,time,record|
+        entries << [time, record]
+      }
+      send_data ['tag1', entries].to_msgpack
```

Comment:
I cannot understand why this code works well... `time` should be an ext typed value, but there are no info about it.